### PR TITLE
OPS: Remove mask_cmp_op fallback behavior

### LINKED ISSUE #28079: API: flex comparisons DataFrame vs Series inconsistency
In tests.frame.test_arithmetic we test the DataFrame part of the following, but we do not test the Series behavior (at least not in that file)

```
arr = np.array([np.nan, 1, 6, np.nan])
arr2 = np.array([2j, np.nan, 7, None])
ser = pd.Series(arr)
ser2 = pd.Series(arr2)
df = pd.DataFrame(ser)
df2 = pd.DataFrame(ser2)

ser < ser2  # raises TypeError, makes sense
df < df2  # raises TypeError, makes sense

ser.lt(ser2)  # raises TypeError, makes sense

>>> df.lt(df2)
       0
0  False
1  False
2   True
3  False
```

The `df.lt(df2)` version (the "flex" op) masks positions where either `df` or `df2` is null.  That is fine, but why doesn't the `Series` version do the same thing?

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #28079 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

xref #28050. 

To make the Series vs DataFrame behavior consistent, the two main options are a) change the DataFrame behavior (this PR) or b) change the Series behavior.  The latter is complicated by the fact that for object dtypes the Series comparisons go through comp_method_OBJECT_ARRAY instead of the numpy op, so we would still have to change the complex-case test changed here.

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. I think this needs a note (you can just put in bug fixes would be fine); as something that 'worked' before now will raise (correctly)

**Comment by jbrockmendel** [comment]
note added

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -199,7 +199,7 @@ Timezones
 Numeric
 ^^^^^^^
 - Bug in :meth:`DataFrame.quantile` with zero-column :class:`DataFrame` incorrectly raising (:issue:`23925`)
--
+- :class:`DataFrame` inequality comparisons with object-dtype and ``complex`` entries failing to raise ``TypeError`` like their :class:`Series` counterparts (:issue:`28079`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.0.rst]: entries -> dtypes (can do in later PR)

**Review by jreback — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@jbrockmendel you mentioned that you changed the behaviour of the flex method for Series or DataFrame (the example in the issue also shows one without complex dtype). Shouldn't that be reflected in the whatsnew note?

**Comment by jorisvandenbossche** [comment]
Updated question: this was only for the comparison *methods* and not the comparison *operation*, correct? If so, I would be clearer on that in the whatsnew note.

**Comment by jbrockmendel** [comment]
> you mentioned that you changed the behaviour of the flex method for Series or DataFrame

`DataFrame`'s `lt, le, gt, ge` methods are affected.  AFAICT the only case where it really makes a difference is with complex-dtypes that used to get cast to object-dtype for masking fallback, because numpy (for some reason) does complex-dtype comparisons differently from python.

Which is a long way of saying I don't understand the question.  Do you have a suggested edit for the whatsnew note?

> this was only for the comparison methods and not the comparison operation, correct? I

I don't understand the distinction.  Can you give an example?

**Comment by jorisvandenbossche** [comment]
> DataFrame's lt, le, gt, ge methods are affected

As you say, that are methods. And with the comparison operators I mean >, <, .. An essential difference in this case, and if you just say "comparison" most people will think about the second IMO

**Comment by jbrockmendel** [comment]
How would you word the note?  Adding the word "flex" is a start.

**Pull request merged**